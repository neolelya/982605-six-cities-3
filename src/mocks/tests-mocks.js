import {City, OFFER_FEATURES, RentalFeature, RentalType} from '../consts';

export const TEST_OFFERS = [
  {
    location: {
      city: City.AMSTERDAM,
      cityCoordinates: [52.38333, 4.9],
    },
    offers: [
      {
        id: 0,
        coordinates: [52.3909553943508, 4.85309666406198],
        rentalHost: {
          hostName: `Angelina`,
          hostAvatar: `img/avatar-angelina.jpg`,
          isSuper: true,
        },
        rentalTitle: `Beautiful & luxurious apartment at great location`,
        rentalImages: [
          `img/apartment-01.jpg`,
          `img/apartment-02.jpg`,
          `img/apartment-03.jpg`,
          `img/apartment-small-03.jpg`,
          `img/apartment-small-04.jpg`,
        ],
        rentalPrice: 120,
        rentalRating: 3.4,
        rentalType: RentalType.APARTMENT,
        isPremium: true,
        isBookmark: true,
        rentalDescription: [
          `A bright and charming apartment with 1 bedroom, located close to Museum Square in one of the most cosmopolitan and vibrant districts of Amsterdam "de PIJP".`,
          `Perfect for short holidays, business trips.`,
        ],
        rentalRoomsQuantity: 3,
        rentalMaxGuestsQuantity: 3,
        rentalFeatures: [RentalFeature.DRYER, RentalFeature.WASHINGMACHINE],
      },
      {
        id: 1,
        coordinates: [52.369553943508, 4.85309666406198],
        rentalHost: {
          hostName: `Max`,
          hostAvatar: `img/avatar-max.jpg`,
          isSuper: false,
        },
        rentalTitle: `Wood and stone place`,
        rentalImages: [
          `img/apartment-02.jpg`,
          `img/apartment-01.jpg`,
          `img/apartment-03.jpg`,
          `img/apartment-small-03.jpg`,
          `img/apartment-small-04.jpg`,
        ],
        rentalPrice: 80,
        rentalRating: 4.1,
        rentalType: RentalType.HOTEL,
        isPremium: false,
        isBookmark: true,
        rentalDescription: [
          `This package is special opportunity to connect accommodation in a traditional Dutch cottage located in a birch forest, taste the local food, and snowshoeing/hiking experience.`,
        ],
        rentalRoomsQuantity: 2,
        rentalMaxGuestsQuantity: 2,
        rentalFeatures: [
          RentalFeature.DRYER,
          RentalFeature.WASHINGMACHINE,
          RentalFeature.CABLETV,
          RentalFeature.KITCHEN,
          RentalFeature.KITCHEN,
        ],
      },
      {
        id: 2,
        coordinates: [52.3909553943508, 4.929309666406198],
        rentalHost: {
          hostName: `Angelina`,
          hostAvatar: `img/avatar-angelina.jpg`,
          isSuper: true,
        },
        rentalTitle: `Stylish and cozy place`,
        rentalImages: [
          `img/apartment-03.jpg`,
          `img/apartment-01.jpg`,
          `img/apartment-02.jpg`,
          `img/apartment-small-03.jpg`,
          `img/apartment-small-04.jpg`,
        ],
        rentalPrice: 132,
        rentalRating: 4.75,
        rentalType: RentalType.HOUSE,
        isPremium: true,
        isBookmark: false,
        rentalDescription: [
          `Spacious apartment situated in the city centre (oud-West) of Amsterdam.`,
          `Walking distance to the Jordaan, Museum Quarter, Anne Frank house and Leidseplein.`,
          `Supermarket right on the corner.`,
        ],
        rentalRoomsQuantity: 4,
        rentalMaxGuestsQuantity: 7,
        rentalFeatures: OFFER_FEATURES,
      },
      {
        id: 3,
        coordinates: [52.3809553943508, 4.939309666406198],
        rentalHost: {
          hostName: `Max`,
          hostAvatar: `img/avatar-max.jpg`,
          isSuper: false,
        },
        rentalTitle: `Excellent location and free parking`,
        rentalImages: [
          `img/room.jpg`,
          `img/room-small.jpg`,
          `img/apartment-01.jpg`,
          `img/apartment-02.jpg`,
          `img/apartment-03.jpg`,
          `img/apartment-small-03.jpg`,
          `img/apartment-small-04.jpg`,
        ],
        rentalPrice: 180,
        rentalRating: 4.5,
        rentalType: RentalType.APARTMENT,
        isPremium: false,
        isBookmark: false,
        rentalDescription: [
          `The studio apartments at Wittenberg have been carefully-designed to embrace and enhance the buildingâ€™s timeless authentic features, whilst incorporating striking yet traditional Dutch interiors.`,
        ],
        rentalRoomsQuantity: 2,
        rentalMaxGuestsQuantity: 4,
        rentalFeatures: [
          RentalFeature.DRYER,
          RentalFeature.WASHINGMACHINE,
          RentalFeature.PARKING,
        ],
      },
    ],
  },
];
