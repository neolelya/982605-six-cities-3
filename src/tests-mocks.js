import {City, OFFER_FEATURES, RentalFeature, RentalType} from './consts';

export const TEST_OFFERS = [
  {
    location: {
      city: City.AMSTERDAM,
      cityCoordinates: [52.38333, 4.9],
    },
    offers: [
      {
        id: 0,
        coordinates: [52.3909553943508, 4.85309666406198],
        rentalHost: {
          hostName: `Angelina`,
          hostAvatar: `img/avatar-angelina.jpg`,
          isSuper: true,
        },
        rentalTitle: `Beautiful & luxurious apartment at great location`,
        rentalImages: [
          `img/apartment-01.jpg`,
          `img/apartment-02.jpg`,
          `img/apartment-03.jpg`,
          `img/apartment-small-03.jpg`,
          `img/apartment-small-04.jpg`,
        ],
        rentalPrice: 120,
        rentalRating: 3.4,
        rentalType: RentalType.APARTMENT,
        isPremium: true,
        isBookmark: true,
        rentalDescription: [
          `A bright and charming apartment with 1 bedroom, located close to Museum Square in one of the most cosmopolitan and vibrant districts of Amsterdam "de PIJP".`,
          `Perfect for short holidays, business trips.`,
        ],
        rentalRoomsQuantity: 3,
        rentalMaxGuestsQuantity: 3,
        rentalFeatures: [RentalFeature.DRYER, RentalFeature.WASHINGMACHINE],
        reviews: [
          {
            id: 0,
            name: `Adam`,
            avatar: `img/avatar-max.jpg`,
            rating: 4,
            date: new Date(2020, 0, 12),
            text: `Easy Check In: Management did extra mile, and allowed me to check in several hours before. Cleaning of my unit was swiftly organized, and properly introduced.`,
          },
          {
            id: 2,
            name: `Alex`,
            avatar: `img/avatar.svg`,
            rating: 5,
            date: new Date(2020, 1, 28),
            text: `We need a quick place to stay for the night and rented the same day. Was nice and clean. Had parking available for 15€ for the night. Was a little noisy but it ended by 11pm and we were able to get a decent night sleep. Had everything we needed.`,
          },
        ],
      },
      {
        id: 1,
        coordinates: [52.369553943508, 4.85309666406198],
        rentalHost: {
          hostName: `Max`,
          hostAvatar: `img/avatar-max.jpg`,
          isSuper: false,
        },
        rentalTitle: `Wood and stone place`,
        rentalImages: [
          `img/apartment-02.jpg`,
          `img/apartment-01.jpg`,
          `img/apartment-03.jpg`,
          `img/apartment-small-03.jpg`,
          `img/apartment-small-04.jpg`,
        ],
        rentalPrice: 80,
        rentalRating: 4.1,
        rentalType: RentalType.HOTEL,
        isPremium: false,
        isBookmark: true,
        rentalDescription: [
          `This package is special opportunity to connect accommodation in a traditional Dutch cottage located in a birch forest, taste the local food, and snowshoeing/hiking experience.`,
        ],
        rentalRoomsQuantity: 2,
        rentalMaxGuestsQuantity: 2,
        rentalFeatures: [
          RentalFeature.DRYER,
          RentalFeature.WASHINGMACHINE,
          RentalFeature.CABLETV,
          RentalFeature.KITCHEN,
          RentalFeature.KITCHEN,
        ],
        reviews: [
          {
            id: 0,
            name: `Adam`,
            avatar: `img/avatar-max.jpg`,
            rating: 4,
            date: new Date(2020, 0, 12),
            text: `Easy Check In: Management did extra mile, and allowed me to check in several hours before. Cleaning of my unit was swiftly organized, and properly introduced.`,
          },
          {
            id: 1,
            name: `Mary`,
            avatar: `img/avatar-angelina.jpg`,
            rating: 5,
            date: new Date(2020, 2, 15),
            text: `The building is in between the beach and metro station, 5 mins walking distance to both; 5 stops away from the Gothic Quarter by metro.`,
          },
          {
            id: 2,
            name: `Alex`,
            avatar: `img/avatar.svg`,
            rating: 5,
            date: new Date(2020, 1, 28),
            text: `We need a quick place to stay for the night and rented the same day. Was nice and clean. Had parking available for 15€ for the night. Was a little noisy but it ended by 11pm and we were able to get a decent night sleep. Had everything we needed.`,
          },
        ],
      },
      {
        id: 2,
        coordinates: [52.3909553943508, 4.929309666406198],
        rentalHost: {
          hostName: `Angelina`,
          hostAvatar: `img/avatar-angelina.jpg`,
          isSuper: true,
        },
        rentalTitle: `Stylish and cozy place`,
        rentalImages: [
          `img/apartment-03.jpg`,
          `img/apartment-01.jpg`,
          `img/apartment-02.jpg`,
          `img/apartment-small-03.jpg`,
          `img/apartment-small-04.jpg`,
        ],
        rentalPrice: 132,
        rentalRating: 4.75,
        rentalType: RentalType.HOUSE,
        isPremium: true,
        isBookmark: false,
        rentalDescription: [
          `Spacious apartment situated in the city centre (oud-West) of Amsterdam.`,
          `Walking distance to the Jordaan, Museum Quarter, Anne Frank house and Leidseplein.`,
          `Supermarket right on the corner.`,
        ],
        rentalRoomsQuantity: 4,
        rentalMaxGuestsQuantity: 7,
        rentalFeatures: OFFER_FEATURES,
        reviews: [
          {
            id: 2,
            name: `Alex`,
            avatar: `img/avatar.svg`,
            rating: 5,
            date: new Date(2020, 1, 28),
            text: `We need a quick place to stay for the night and rented the same day. Was nice and clean. Had parking available for 15€ for the night. Was a little noisy but it ended by 11pm and we were able to get a decent night sleep. Had everything we needed.`,
          },
        ],
      },
      {
        id: 3,
        coordinates: [52.3809553943508, 4.939309666406198],
        rentalHost: {
          hostName: `Max`,
          hostAvatar: `img/avatar-max.jpg`,
          isSuper: false,
        },
        rentalTitle: `Excellent location and free parking`,
        rentalImages: [
          `img/room.jpg`,
          `img/room-small.jpg`,
          `img/apartment-01.jpg`,
          `img/apartment-02.jpg`,
          `img/apartment-03.jpg`,
          `img/apartment-small-03.jpg`,
          `img/apartment-small-04.jpg`,
        ],
        rentalPrice: 180,
        rentalRating: 4.5,
        rentalType: RentalType.APARTMENT,
        isPremium: false,
        isBookmark: false,
        rentalDescription: [
          `The studio apartments at Wittenberg have been carefully-designed to embrace and enhance the building’s timeless authentic features, whilst incorporating striking yet traditional Dutch interiors.`,
        ],
        rentalRoomsQuantity: 2,
        rentalMaxGuestsQuantity: 4,
        rentalFeatures: [
          RentalFeature.DRYER,
          RentalFeature.WASHINGMACHINE,
          RentalFeature.PARKING,
        ],
        reviews: [],
      },
    ],
  },
];

export const TEST_REVIEWS = [
  {
    id: 0,
    name: `Adam`,
    avatar: `img/avatar-max.jpg`,
    rating: 4,
    date: new Date(2020, 0, 12),
    text: `Easy Check In: Management did extra mile, and allowed me to check in several hours before. Cleaning of my unit was swiftly organized, and properly introduced.`,
  },
  {
    id: 1,
    name: `Mary`,
    avatar: `img/avatar-angelina.jpg`,
    rating: 5,
    date: new Date(2020, 2, 15),
    text: `The building is in between the beach and metro station, 5 mins walking distance to both; 5 stops away from the Gothic Quarter by metro.`,
  },
  {
    id: 2,
    name: `Alex`,
    avatar: `img/avatar.svg`,
    rating: 5,
    date: new Date(2020, 1, 28),
    text: `We need a quick place to stay for the night and rented the same day. Was nice and clean. Had parking available for 15€ for the night. Was a little noisy but it ended by 11pm and we were able to get a decent night sleep. Had everything we needed.`,
  },
];

export const TEST_OFFERS_COORDINATES = [
  [52.3809553943508, 4.939309666406198],
  [52.3909553943508, 4.929309666406198],
  [52.3909553943508, 4.85309666406198],
  [52.369553943508, 4.85309666406198],
];
