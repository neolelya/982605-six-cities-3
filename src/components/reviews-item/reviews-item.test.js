import React from 'react';
import renderer from 'react-test-renderer';
import ReviewsItem from './reviews-item.jsx';
import {REVIEWS} from '../../tests-mocks';

it(`Should render ReviewsItem correctly`, () => {
  const tree = renderer.create(<ReviewsItem review={REVIEWS[0]} />).toJSON();

  expect(tree).toMatchSnapshot();
});
