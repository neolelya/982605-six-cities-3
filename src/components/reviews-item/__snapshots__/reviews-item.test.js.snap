// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ReviewsItem correctly 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        src="img/avatar-max.jpg"
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      Adam
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      Easy Check In: Management did extra mile, and allowed me to check in several hours before. Cleaning of my unit was swiftly organized, and properly introduced.
    </p>
    <time
      className="reviews__time"
      dateTime="2020-01-12"
    >
      January 12, 2020
    </time>
  </div>
</li>
`;
